### DISCUSSION TITLE
MQTT persistent session

### DISCUSSION THREAD

**Discussion by DetlefR** [arendst/Tasmota#15519] [Q&A] [answered]
Hello,

is it possible to have a SetOption??? for the MQTT_CLEAN_SESSION flag. I have some (not permanent powered) devices, they need  also messages if they are not online.
And every time build a separate firmware makes the maintenance not easier.

Thanks
Detlef

**Comment by barbudor** [comment, marked as answer]
From my understanding (but I may be wrong), in order for this to work QoS = 1 must be used on both sender and receiver.
As Tasmota's PubSubClient library only support QoS=0, I fear that Tasmota will not receive messages even if persistent session (cleanSesstion = false) was used.

At least this is what the table at the end of this article suggest: http://www.steves-internet-guide.com/mqtt-clean-sessions-example/

![image](https://user-images.githubusercontent.com/2996042/166332778-a569a310-a37e-44f5-a125-e1f7608bc6b2.png)

**Comment by DetlefR** [comment]
Hello,

sure, it needs a little effort to make it working. Tasmota doesn't set a subscription for the qos1 queue automatically. I must do it manually in the broker(I use rabbitmq). But after them, it is working.
Unfortunately, sometimes I forget that this device has a special firmware and then is an update done where it should not be.

**Reply by barbudor** [reply]
You can also automate quite easily the build of custom firmware.
That's what I'm doing as I have a nighly build of a few Tasmota variants with my own settings (Wifi credentials of course + a few other options) which allows my device to always reconnect easily to my network even in case of full erase.

**Reply by barbudor** [reply]
Here you are
https://github.com/arendst/Tasmota/pull/15530
